From d9ed26f7b28a254b3e2d365bb0209a80bfbead36 Mon Sep 17 00:00:00 2001
From: "Guillaume W. Bres" <guillaume.bressaix@gmail.com>
Date: Thu, 9 Dec 2021 20:55:21 +0100
Subject: [PATCH 4/5] package/pybind11

package/pybind11: new package

Pybind11 is a lightweight header-only library
to create C++ bindings of existing Python code.

This is the C++ side of package "python-pybind".

Signed-off-by: Guillaume W. Bres <guillaume.bressaix@gmail.com>
---
Installation to target does not make sense for such a package.
Header files can be installed to staging and / or host.

Signed-off-by: Guillaume W. Bres <guillaume.bressaix@gmail.com>
---
 DEVELOPERS                     |  1 +
 package/Config.in              |  1 +
 package/pybind11/Config.in     |  8 ++++++++
 package/pybind11/pybind11.hash |  4 ++++
 package/pybind11/pybind11.mk   | 33 +++++++++++++++++++++++++++++++++
 5 files changed, 47 insertions(+)
 create mode 100644 package/pybind11/Config.in
 create mode 100644 package/pybind11/pybind11.hash
 create mode 100644 package/pybind11/pybind11.mk

diff --git a/DEVELOPERS b/DEVELOPERS
index 3023526..7a0bb76 100644
--- a/DEVELOPERS
+++ b/DEVELOPERS
@@ -1105,6 +1105,7 @@ F:	package/libnids/
 F:	package/libxcrypt/
 F:	package/liquid-dsp/
 F:	package/pixiewps/
+F:	package/pybind11/
 F:	package/python-pybind/
 F:	package/reaver/
 
diff --git a/package/Config.in b/package/Config.in
index 5720830..c9aa15b 100644
--- a/package/Config.in
+++ b/package/Config.in
@@ -2015,6 +2015,7 @@ endif
 	source "package/protobuf/Config.in"
 	source "package/protobuf-c/Config.in"
 	source "package/protozero/Config.in"
+	source "package/pybind11/Config.in"
 	source "package/qhull/Config.in"
 	source "package/qlibc/Config.in"
 	source "package/riemann-c-client/Config.in"
diff --git a/package/pybind11/Config.in b/package/pybind11/Config.in
new file mode 100644
index 0000000..cdc81d8
--- /dev/null
+++ b/package/pybind11/Config.in
@@ -0,0 +1,8 @@
+config BR2_PACKAGE_PYBIND11
+	bool "pybind11"
+	help
+	  Pybind11 is a lightweight header-only library
+	  to create C++ bindings of existing python code.
+	  This is the C++ side of package "python-pybind".
+
+	  http://pybind11.readthedocs.org/en/master
diff --git a/package/pybind11/pybind11.hash b/package/pybind11/pybind11.hash
new file mode 100644
index 0000000..531b2c4
--- /dev/null
+++ b/package/pybind11/pybind11.hash
@@ -0,0 +1,4 @@
+# Locally calculated
+sha256  f1bcc07caa568eb312411dde5308b1e250bd0e1bc020fae855bf9f43209940cc  pybind11-2.8.1.tar.gz
+# License files, locally calculated
+sha256  83965b843b98f670d3a85bd041ed4b372c8ec50d7b4a5995a83ac697ba675dcb  LICENSE
diff --git a/package/pybind11/pybind11.mk b/package/pybind11/pybind11.mk
new file mode 100644
index 0000000..ab0e7c1
--- /dev/null
+++ b/package/pybind11/pybind11.mk
@@ -0,0 +1,33 @@
+################################################################################
+#
+# pybind11
+#
+################################################################################
+
+PYBIND11_VERSION = 2.8.1
+PYBIND11_SITE = $(call github,pybind,pybind11,v$(PYBIND11_VERSION))
+PYBIND11_LICENSE = BSD-3-Clause
+PYBIND11_LICENSE_FILES = LICENSE
+PYBIND11_INSTALL_TARGET = NO
+PYBIND11_INSTALL_STAGING = YES
+PYBIND11_SUPPORTS_IN_SOURCE_BUILD = YES
+
+PYBIND11_CONF_OPTS += \
+	-DBUILD_DOCS=OFF \
+	-DPYBIND11_TEST=OFF
+
+HOST_PYBIND11_CONF_OPTS += \
+	-DBUILD_DOCS=OFF \
+	-DPYBIND11_TEST=OFF
+
+# use 'find python' feature with dedicated variables
+# to avoid /usr/bin/python possibly preferred
+PYBIND11_CONF_OPTS += \
+	-DPYBIND11_FINDPYTHON=ON \
+	-DPython_EXECUTABLE=$(HOST_DIR)/usr/bin/python
+HOST_PYBIND11_CONF_OPTS += \
+	-DPYBIND11_FINDPYTHON=ON \
+	-DPython_EXECUTABLE=$(HOST_DIR)/usr/bin/python
+
+$(eval $(cmake-package))
+$(eval $(host-cmake-package))
-- 
1.8.3.1

